x/tax2gas/post/post.go

```diff
@@ -154,13 +127,15 @@ func (tgd Tax2gasPostDecorator) PostHandle(ctx sdk.Context, tx sdk.Tx, simulate
 
 		return ctx, errorsmod.Wrapf(sdkerrors.ErrInsufficientFee, "fees are not enough to pay for gas, need to cover %s gas more, which equal to %q ", gasRemaining.String(), gasRemainingFees)
 	}
-	feePayerAccount := tgd.accountKeeper.GetAccount(ctx, feePayer)
-	err := tgd.bankKeeper.SendCoinsFromAccountToModule(ctx, feePayerAccount.GetAddress(), authtypes.FeeCollectorName, payableFees)
-	if err != nil {
-		return ctx, errorsmod.Wrapf(sdkerrors.ErrInsufficientFunds, err.Error())
+	deductFeesFromAcc := tgd.accountKeeper.GetAccount(ctx, deductFeesFrom)
+	if !feeCoins.IsZero() {
+		err := tgd.bankKeeper.SendCoinsFromAccountToModule(ctx, deductFeesFromAcc.GetAddress(), authtypes.FeeCollectorName, feeCoins)
```

Comment:
There is no way you can do that, you only have 2 options: pay all the fees (which tax is converted to gas) or pay with exact amount that the chain need.

Reply:
What you define will not gonna work, cause the coin you trying to deduct will still less than the tx fee that user put in, so that will not be the same logic as cosmos sdk do.

Reply:
What you do is just trying to make user pay more money than it actually needed.